base/linalg/arpack.jl

```diff
@@ -228,7 +228,7 @@ for (T, saupd_name, seupd_name, naupd_name, neupd_name) in
                   sizeof(howmny), sizeof(bmat), sizeof(evtype))
         end
 
-        function saupd(ido, bmat, n, which, nev, TOL::Array{$T}, resid::Array{$T}, ncv, v::Array{$T}, ldv,
+        function saupd(ido, bmat, n, which, nev, TOL::Ref{$T}, resid::Array{$T}, ncv, v::Array{$T}, ldv,
```

Comment:
I guess, although it seems unnecessarily complex. I think the first suggestion to just revert and then refactor later after 0.6 might be the best course of action.